### DISCUSSION TITLE
Helper that returns HTML inside each block ( c#)

### DISCUSSION THREAD

**Discussion by shahab0105** [handlebars-lang/handlebars.js#1909] [Q&A]
I have a handlebar template which renders html dynamically based on a data array provided, now this content sometimes spans over multiple pages and since I'm trying to generate a pdf based of this html it hides the overflowing context instead of contiuning the pdf on a new page( IText7). After spending quite sometime ive realized this can be resolved with the help of some handlebar helpers. 

I am trying to add pagebreaks/page change in my handlebar template, and since these pagebreaks are going to be on every nth element (because only n elements would fit on a pdf page), id need a helper that returns html on every nth index inside an each loop. I've gotten somewhat close, here the helper
```
handlebars.RegisterHelper("changePage", (writer, context, arguments) =>
            {

                if (Convert.ToInt32(arguments[0]) > 0 && Convert.ToInt32(arguments[0]) % 3 == 0)
                {
                    
                    writer.Write($"<div class='pageBreak'> </div>");

                }
               
            });
```

and using it inside an each would be 

```
  {{#each portfolioDataPage2}}
            {{changePage @index}}
            
```

The only problem is that it escapes the html while i would want to inject html.  Trying writer.WriteSafeString("<div></div>") would yield an error because writesafestring based helper cant be used inside an each block helper for some reason (came to this conclusion because thewritesafestring helper would work outside of each block). ps i already have css for pageBreak class in my template so just adding an empty element with that class would add these pagebreaks for me 
for anyone curious this css results in a pagebreak 

```
.pageBreak{
      page-break-before: always;
    }
```

Can anyone guide me how to create a helper in c# that returns html and would work inside an each block?

**Comment by jaylinski** [comment]
> Can anyone guide me how to create a helper in c#

For C# questions, please refer to the .NET handlebars-implementation: https://github.com/Handlebars-Net/Handlebars.Net
